pkg/actuators/machine/actuator_test.go

```diff
@@ -3,184 +3,204 @@ package machine
+				awsClientBuilder = func(client runtimeclient.Client, secretName, namespace, region string) (awsclient.Client, error) {
+					return nil, errors.New("AWS client error")
+				}
 			}
 
 			if tc.awsError {
-				mockAWSClient.EXPECT().DescribeInstances(gomock.Any()).Return(nil, errors.New("error")).AnyTimes()
+				mockAWSClient.EXPECT().DescribeInstances(gomock.Any()).Return(nil, errors.New("AWS error")).AnyTimes()
 			} else {
-
 				mockAWSClient.EXPECT().DescribeInstances(gomock.Any()).Return(stubDescribeInstancesOutput("ami-a9acbbd6", "i-02fcb933c5da7085c", ec2.InstanceStateNameRunning), nil).AnyTimes()
 			}
 
-			mockAWSClient.EXPECT().RunInstances(gomock.Any()).Return(stubReservation("ami-a9acbbd6", "i-02fcb933c5da7085c"), tc.runInstancesErr).AnyTimes()
+			mockAWSClient.EXPECT().RunInstances(gomock.Any()).Return(stubReservation("ami-a9acbbd6", "i-02fcb933c5da7085c"), nil).AnyTimes()
 			mockAWSClient.EXPECT().TerminateInstances(gomock.Any()).Return(&ec2.TerminateInstancesOutput{}, nil)
 			mockAWSClient.EXPECT().RegisterInstancesWithLoadBalancer(gomock.Any()).Return(nil, nil).AnyTimes()
-			mockAWSClient.EXPECT().TerminateInstances(gomock.Any()).Return(&ec2.TerminateInstancesOutput{}, tc.terminateInstancesErr).AnyTimes()
-			mockAWSClient.EXPECT().RegisterInstancesWithLoadBalancer(gomock.Any()).Return(nil, tc.lbErr).AnyTimes()
-			mockAWSClient.EXPECT().ELBv2DescribeLoadBalancers(gomock.Any()).Return(stubDescribeLoadBalancersOutput(), tc.lbErr)
+			mockAWSClient.EXPECT().TerminateInstances(gomock.Any()).Return(&ec2.TerminateInstancesOutput{}, nil).AnyTimes()
+			mockAWSClient.EXPECT().RegisterInstancesWithLoadBalancer(gomock.Any()).Return(nil, nil).AnyTimes()
+			mockAWSClient.EXPECT().ELBv2DescribeLoadBalancers(gomock.Any()).Return(stubDescribeLoadBalancersOutput(), nil).AnyTimes()
 			mockAWSClient.EXPECT().ELBv2DescribeTargetGroups(gomock.Any()).Return(stubDescribeTargetGroupsOutput(), nil).AnyTimes()
 			mockAWSClient.EXPECT().ELBv2RegisterTargets(gomock.Any()).Return(nil, nil).AnyTimes()
 
+			params := ActuatorParams{
+				Client:           k8sClient,
+				EventRecorder:    eventRecorder,
+				AwsClientBuilder: awsClientBuilder,
+			}
 			actuator := NewActuator(params)
+			tc.operation(actuator, machine)
 
-			tc.operation(actuator, tc.machine)
-			select {
-			case event := <-eventsChannel:
-				if event != tc.event {
-					t.Errorf("Expected %q event, got %q", tc.event, event)
+			eventList := &v1.EventList{}
+			waitForEvent := func() error {
+				g.Expect(k8sClient.List(ctx, eventList, client.InNamespace(machine.Namespace))).To(Succeed())
+				if len(eventList.Items) != 1 {
+					errorMsg := fmt.Sprintf("Expected len 1, got %d", len(eventList.Items))
+					return errors.New(errorMsg)
 				}
-			default:
-				t.Errorf("Expected %q event, got none", tc.event)
+				return nil
+			}
+
+			g.Eventually(waitForEvent, timeout).Should(Succeed())
+
+			g.Expect(tc.event).To(Equal(eventList.Items[0].Message))
```

Comment:
nit: This expectation is the wrong way round, conventionally you say you expect the thing you get from the API to equal the test case expected object
```suggestion
			g.Expect(eventList.Items[0].Message).To(Equal(tc.event))
```